cmake/rust.cmake

```diff
@@ -0,0 +1,59 @@
+include(ExternalProject)
+set_directory_properties(PROPERTIES EP_PREFIX ${CMAKE_BINARY_DIR}/rust)
+
+ExternalProject_Add(
+    rustls-ffi
+    DOWNLOAD_COMMAND ""
+    CONFIGURE_COMMAND ""
+    BUILD_COMMAND
+        cargo capi build --locked ${CARGO_FEATURES}
+        "$<IF:$<CONFIG:Release>,--release,-->"
+    # Rely on cargo checking timestamps, rather than tell CMake where every
+    # output is.
+    BUILD_ALWAYS true
+    INSTALL_COMMAND
+        cargo capi install --libdir=lib --prefix=${CMAKE_BINARY_DIR}/rust
+        --locked ${CARGO_FEATURES} "$<IF:$<CONFIG:Release>,--release,-->"
```

Comment:
one thing to improve here: we're building too much.

Previously the Windows CMake had `BUILD_ALWAYS true` with the associated comment, but used `cargo` as a `COMMAND` and left the `BUILD_COMMAND` empty. 

I don't think the choice of `COMMAND` vs `BUILD_COMMAND` is load bearing here (?) but I think `cargo capi` is building even when there aren't changes in the Rust code. Combined with `BUILD_ALWAYS true` this means we build everything for each target which is a bummer.

I'll give this some thought.

Reply:
> Or maybe cargo-c and an IDE were fighting over one hash key in the target directory?

I think that's a likely culprit. I haven't been able to reproduce this locally once I closed CLion and made sure my `cargo-c` install was up to date. I was mostly building on CLI but had also set up the IDE integration and suspect that's a recipe for rebuilds.

Thanks for checking!